blueprints/component/files/__root__/__path__/__name__.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Component definition for the <%= dasherizedModuleName %> component
+ */
+
+import Ember from 'ember'
+const {Component} = Ember
+import PropTypesMixin, {PropTypes} from 'ember-prop-types'
+import computed, {readOnly} from 'ember-computed-decorators'
+
+import layout from '<%= templatePath %>'
+
+export default Component.extend(PropTypesMixin, {
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+
+  /**
+   * Properties for this component. Public properties are expected to be
+   * (potentially) passed in to the component. Private properties are *not*
+   * expected to be passed in/overwritten.
+   */
+  propTypes: {
+    // public
+    hook: PropTypes.string,
```

Comment:
Why? We have a default value for it. Making it required means that anytime anyone uses your component, they have to specify a hook. I know that for some components, ones I don't use more than once in a route, I don't normally set the hook, I just use the default.